conda_forge_tick/new_update_versions.py

```diff
@@ -70,35 +73,34 @@ def new_update_upstream_versions(
 
     # Inspection the graph object and node update:
     # print(f"Number of nodes: {len(gx.nodes)}")
-    Node_count = 0
-
+    node_count = 0
     to_update = []
     for node, node_attrs in tqdm.tqdm(_all_nodes):
-        attrs = node_attrs["payload"]
-        if attrs.get("bad") or attrs.get("archived"):
-            attrs["new_version"] = False
-            continue
-        to_update.append((node, attrs))
+        with node_attrs["payload"] as attrs:
+            if attrs.get("bad") or attrs.get("archived"):
+                # "new_version" = False
+                continue
+            to_update.append((node, attrs))
 
-    up_to = {}
     for node, node_attrs in to_update:
         # checking each node
         with node_attrs as attrs:
             up_to = {}
 
-            # verify the actual situation of the package;
-            actual_ver = str(attrs.get("version"))
-
-            # rude exception
+            # avoid
             if node == "ca-policy-lcg":
-                up_to["ca-policy-lcg"] = {"bad": attrs.get("bad"), "new_version": False}
-                Node_count += 1
+                up_to["new_version"] = False
+                node_count += 1
                 continue
 
             # New version request
             try:
-                new_version = get_latest_version(node, attrs, sources)
-                attrs["new_version"] = new_version or attrs["new_version"]
+                # check for latest version
+                new_version = (
+                    get_latest_version(node, attrs, sources, up_to)
+                    or attrs["new_version"]
```

Comment:
idk if we want to use the existing `new_version` if we don't get a version then the file should say so.

Reply:
Could it be replaced by False ?

Reply:
I would have it return an empty dict or not write out the file at all (if it would be empty).

Reply:
I've added a dict containing the `ver` output and `bad` error string (when it occurs) inside `get_latest_version`. Then update using `version_data.update(get_latest_version(args))`